test/e2e_federation/framework/cluster.go

```diff
@@ -0,0 +1,164 @@
+/*
```

Comment:
Is it worth basing this off of the file that has much of its source? It would be nice not to lose the history.